plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueMetastoreModule.java

```diff
@@ -89,22 +89,6 @@ public Executor createExecutor(CatalogName catalogName, GlueHiveMetastoreConfig
         return createExecutor("hive-glue-partitions-%s", hiveConfig.getGetPartitionThreads());
     }
 
-    @Provides
-    @Singleton
-    @ForGlueColumnStatisticsRead
-    public Executor createStatisticsReadExecutor(CatalogName catalogName, GlueHiveMetastoreConfig hiveConfig)
-    {
-        return createExecutor("hive-glue-statistics-read-%s", hiveConfig.getReadStatisticsThreads());
-    }
-
-    @Provides
-    @Singleton
-    @ForGlueColumnStatisticsWrite
-    public Executor createStatisticsWriteExecutor(CatalogName catalogName, GlueHiveMetastoreConfig hiveConfig)
-    {
-        return createExecutor("hive-glue-statistics-write-%s", hiveConfig.getWriteStatisticsThreads());
-    }
-
     private Executor createExecutor(String nameTemplate, int threads)
```

Comment:
On the second thought, I am not sure that dropping those executors in favor of using async methods is a great idea. 
We are losing fine-grained control over how many threads concurrent requests we allow for statistics writes vs statistic reads.

If we do so probably we should use multiple glueClient instances for different tasks, and configure them differently. 

I suggest we drop this commit from this PR and keep the change minimal, focusing on a loop in `addPartitions` (second commit). 
Sorry for back and forth :/

Reply:
The read and write thread pool count default to 1 right now anyway, so who knows if anyone actually sets those. Having separate clients for separate tasks sounds reasonable though. I'll take that commit out and we can look at that separately.